### ISSUE TITLE
Where query produces unnecessary extra query

### ISSUE THREAD

**Issue by jameskleeh** [grails/grails-data-mapping#989]
### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Run the application
2. Visit http://localhost:8080/author/index
3. Inspect the console output after "RELEVANT SQL HERE"

### Expected Behaviour

A query for the book should be executed, then a single query for the authors

### Actual Behaviour

The query for the authors looks correct, then another query is executed to retrieve the books again, even though they were joined in the previous query.

### Environment Information

- **Operating System**: macOS
- **GORM Version:** 6.1.6
- **Grails Version (if using Grails):** 3.3
- **JDK Version:** 1.8

### Example Application

- https://github.com/jameskleeh/where-extra-query

**Comment by graemerocher** [comment]
I don't believe this is a bug, the query is limiting the number of books returned by id which means that only a single book is returned in the result that queries for authors, so in order for Hibernate to obtain all of the books a second query is needed.
